### PR TITLE
Add Mars protocol on Osmosis

### PR DESCRIPTION
**Pull request by larry0x**
Mars protocol was previously been shutdown after the Terra crash, at which time we made a PR to set the protocol's TVL to zero: https://github.com/DefiLlama/DefiLlama-Adapters/pull/2787

Just now, Mars is relaunched on Osmosis. This PR adds TVL tracking for the lending market on Osmosis.

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/mars: 
        
 
  
    
    
    Error: 
        Unknown chain(s): mars
        Note: if you think that the chain is correct but missing from our list, please add it to 'projects/helper/chains.json' file

**Comment by llamatester** [comment]
The adapter at projects/mars exports TVL: 
        
 
 
    terra                     0
    terra-borrowed            0
    mars                      0
    mars-staking              0
    borrowed                  0
    staking                   0
    
    total                    0

File: projects/mars/index.js [hunk 1]
```diff
@@ -1,13 +1,60 @@
...
+const BigNumber = require("bignumber.js");
+
 const zero = (timestamp, block) => ({});
 
+/**
+ * Query the total amount of MARS tokens staked on Mars Hub app-chain.
+ */
+const queryBondedTokens = async (timestamp, block) => {
+  const res = await axios.get(
+    "https://rest.marsprotocol.io/cosmos/staking/v1beta1/pool"
```
[the remark this answers is not in the document]
Comment (larry0x) [reply projects/mars/index.js:11]: Last time I checked, there is a "staking" category for staked tokens?

**Comment by llamatester** [comment]
The adapter at projects/mars exports TVL: 
        
 
 
    terra                     0
    terra-borrowed            0
    osmosis-borrowed          13.012999999999998
    mars                      0
    osmosis                   51.95 k
    mars-staking              0
    borrowed                  13.012999999999998
    staking                   0
    
    total                    51.95 k

**Comment by llamatester** [comment]
The adapter at projects/mars exports TVL: 
        
 
 
    terra                     0
    terra-borrowed            0
    osmosis-borrowed          32.704715
    osmosis                   35.82 k
    borrowed                  32.704715
    
    total                    35.82 k

**Comment by llamatester** [comment]
The adapter at projects/mars exports TVL: 
        
 
 
    terra                     0
    terra-borrowed            0
    osmosis                   35.82 k
    osmosis-borrowed          32.704715
    borrowed                  32.704715
    
    total                    35.82 k

**Comment by llamatester** [comment]
The adapter at projects/mars exports TVL: 
        
 
 
    terra                     0
    terra-borrowed            0
    osmosis-borrowed          32.704715
    osmosis                   35.82 k
    borrowed                  32.704715
    
    total                    35.82 k

File: projects/mars/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (larry0x) [reply projects/mars/index.js:20]: @g1nt0ki I've removed MARS token staking from the TVL

**Comment by llamatester** [comment]
The adapter at projects/mars exports TVL: 
        
 
 
    terra                     0
    terra-borrowed            0
    osmosis                   35.84 k
    osmosis-borrowed          32.694715
    borrowed                  32.694715
    
    total                    35.84 k

File: projects/mars/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (g1nt0ki) [reply projects/mars/index.js]: > Last time I checked, there is a "staking" category for staked tokens?

this is usually for protocols, not chains

[the remark this answers is not in the document]
Comment (larry0x) [reply projects/mars/index.js]: > this is usually for protocols, not chains

Mars is both a protocol AND a chain. The staked tokens are used in governance voting just like AAVE, CRV, SUSHI, etc. but with the additional function of securing the chain.

**Comment by g1nt0ki** [comment]
hi @larry0x , I refactored it a bit here: https://github.com/DefiLlama/DefiLlama-Adapters/pull/5267 

We dont count all supply as tvl but liquidity that is not yet lent out. Maybe it is better to have a new listing called 'Mars - Redbank', it will be listed as sub page of mars protocol, what do you think?

**Comment by g1nt0ki** [comment]
@larry0x thanks for the PR (and review), closing this, fixed it here: https://github.com/DefiLlama/DefiLlama-Adapters/commit/78dd286010c7e34f0f79264df042c3ed767d1cde

**Comment by larry0x** [comment]
@g1nt0ki Thank you!

**Pull request closed without merging**